### DISCUSSION TITLE
Add ability to remove a package outside of admin

### DISCUSSION THREAD

**Discussion by ghost** [djangopackages/djangopackages#992] [Feature Ideas]
Greetings,

I added a package by mistake: https://djangopackages.org/packages/p/dotfiles/

I don't see any button to delete, is it possible to do that?

I am sorry, I just wanted to see how the website is working (I am thinking about developing a clone for another web framework).

**Comment by ryancheley** [comment]
Hello @carlotm 

If you’re just looking for a way to poke around the project without accidentally adding a package you can check out the [Quick Start guide](https://github.com/djangopackages/djangopackages#quickstart). The [Docs](https://docs.djangopackages.org/en/latest/install/) also have a more detailed set of instructions for local development. 

@jefftriplett the only place I can find to delete a package is in the admin. There is a [function](https://github.com/djangopackages/djangopackages/blob/a0b3abf0f4add2752001360a8a1f02f7ba751325/grid/views.py#L166) to delete a package from a grid, but there doesn’t seem to be a way to remove the actual package. I see a few issues were packages needed to be deleted (#535 and #881 for example) but it looks like they’ve always been deleted using the admin. 

Should there be a way for a user to delete a package that they added?

**Comment by jefftriplett** [comment]
@carlotm thanks for opening the issue. I saw this earlier today but hadn't had time to follow up. It's removed now. 

@ryancheley I normally remove these from the admin, but I do we had either a status field or an `active: bool = True` field so we can pull things. 

Not sure if something like this would make sense.

Quoting form chatgpt and not 100% relevant to testing publishing a package: 

```python
from django.db import models
from django_fsm import FSMField, transition

class Package(models.Model):
    # Other fields

    # FSM
    STATE_CHOICES = (
        ('new', 'New'),
        ('under_review', 'Under Review'),
        ('accepted', 'Accepted'),
        ('rejected', 'Rejected'),
    )

    REJECT_REASON_CHOICES = (
        ('irrelevant', 'Irrelevant'),
        ('spam', 'Spam'),
        ('unmaintained', 'Unmaintained'),
    )

    state = FSMField(default='new', choices=STATE_CHOICES, protected=True)
    reject_reason = models.CharField(max_length=20, choices=REJECT_REASON_CHOICES, blank=True, null=True)

    @transition(field=state, source='new', target='under_review')
    def review(self):
        pass  # You can add additional actions here if needed

    @transition(field=state, source='under_review', target='accepted')
    def accept(self):
        pass  # Additional actions

    @transition(field=state, source='under_review', target='rejected')
    def reject_for_irrelevance(self):
        self.reject_reason = 'irrelevant'

    @transition(field=state, source='under_review', target='rejected')
    def reject_for_spam(self):
        self.reject_reason = 'spam'

    @transition(field=state, source='under_review', target='rejected')
    def reject_for_unmaintained(self):
        self.reject_reason = 'unmaintained'
```

**Comment by ryancheley** [comment]
@jefftriplett this seems like a really good idea but would all packages need to be reviewed first? 

Maybe the ability to move it from a default of accepted to one of the rejection reasons 🤔

```python

    @transition(field=state, source='new', target='accepted')
    def accept(self):
        pass  # Additional actions

    @transition(field=state, source='accepted', target='rejected')
    def reject_for_irrelevance(self):
        self.reject_reason = 'irrelevant'

    @transition(field=state, source='accepted', target='rejected')
    def reject_for_spam(self):
        self.reject_reason = 'spam'

    @transition(field=state, source='accepted', target='rejected')
    def reject_for_unmaintained(self):
        self.reject_reason = 'unmaintained'
```

Should we move this over to a discussion to determine some of the finer points of a potential implementation?

**Comment by jefftriplett** [comment]
@ryancheley a discussion sounds good. Feel free and I'll close this issue out.

**Comment by ryancheley** [comment]
@jefftriplett would there be any benefit to having a scoring algorithm of some kind that will flag something as needing review? I really like the idea of having states associated with packages that have been added and think that could be really useful. I'm not sure all of the items that would go into a scoring algorithm, but there's probably more than enough data to support it. 

If not that then maybe just the ability for an admin to move update the state from the front end instead of needing to go into the Admin to remove it (or update the state)
